### PR TITLE
Fix PWA, .webmanifest, generic.py

### PR DESCRIPTION
**Pull request by mrbpp**
Hello, guys!

This is my first commit, and I am very happy to be part of this great community and contribute. I found these errors in the Progressive Web App (PWA), so I decided to fix them.

screenshot_00
screenshot_01

I use [Maskable App](https://maskable.app/) to create maskable icons with the classic LNbits color.

screenshot_02
screenshot_03

I wasn't sure which sizes you needed, so I chose some of them.

screenshot_04

I decided to add them in the 'static directory' where the favicon file was.

screenshot_05

Source: [Mozilla PWA](https://developer.mozilla.org/en-US/docs/Web/Manifest/shortcuts)

screenshot_06

No more PWA errors; everything works fine now.

screenshot_07

The new icons look great.

screenshot_08

Maybe you will decide to create new official PWA screenshots from a different page.

screenshot_09

When a user decides to install the PWA, they will see the screenshot_desktop (wide) if they are on a desktop, and if they are on a phone, they will see the screenshot_phone (narrow). I highly recommend changing the description from 'Bitcoin Lightning Wallet' to 'LNbits - free and open-source lightning wallet'; it will look better in my opinion.

screenshot_10

Maskable icon after installing the app on my Mac.

screenshot_11

The screenshots aren't the best.

screenshot_desktop
screenshot_phone

**P.S.** Happy first commit to me!

### CODE REVIEW

**Review by prusnak — CHANGES_REQUESTED** [review]
1. Please put images to `lnbits/static/images` where other images are.

2. Please install optipng and run `tools/optipng.sh` script to optimize PNGs

**Comment by mrbpp** [comment]
Hey @prusnak,

I've successfully wrapped up the tasks you assigned me:

1. I've relocated the four `.png` files to the `lnbits/static/images` directory, along with the two `.webp` files.

2. I've optimized the `PNGs` using `OptiPNG`, and here's the detailed report:

   - **lnbits/static/images/maskable_icon.png**
     - Original: 50859 bytes
     - Optimized: 22401 bytes **(55.95% decrease)**

   - **lnbits/static/images/maskable_icon_x192.png**
     - Original: 5647 bytes
     - Optimized: 3490 bytes **(38.20% decrease)**

   - **lnbits/static/images/maskable_icon_x512.png**
     - Original: 19027 bytes
     - Optimized: 9345 bytes **(50.89% decrease)**

   - **lnbits/static/images/maskable_icon_x96.png**
     - Original: 2545 bytes
     - Optimized: 1522 bytes **(40.20% decrease)**

Tasks completed, and the `PNGs` are now optimized for smoother performance.

I haven't optimized the two .webp screenshots yet. I'm thinking we might want to consider using a tool specifically tailored for `WebP` images, such as `cwebp`. To streamline the process, we could either create a new script, similar to `optipng.sh`, dedicated to `.webp` images, or we can extend the existing script to handle this format as well. I'm not entirely certain about the best approach.


```
{
                "src": (
                    settings.lnbits_custom_logo
                    if settings.lnbits_custom_logo
                    else "https://cdn.jsdelivr.net/gh/lnbits/lnbits@main/docs/logos/lnbits.png"
                ),
                "sizes": "512x512",
                "type": "image/png"
            },
            {
      "src": (
                    settings.lnbits_custom_favicon
                    if settings.lnbits_custom_favicon
                    else "static/favicon.ico"
                ),
      "sizes": "32x32",
      "type": "image/x-icon"
    },
```

We can enhance the configurational flexibility by adding extra settings for different images like `settings.lnbits_custom_logo`. For instance, we might introduce settings such as `settings.lnbits_custom_favicon`, `settings.lnbits_custom_maskable_icon` ... ,  and so forth. This approach would enable seamless customization of various images across the entire application.

PWA tested on Android phone - Screenshots:

screenshot_12
screenshot_13
screenshot_15

New bugs were found in the PWA on the phone:
- Custom messages are in front of the Logo Text/Logo Icon.
- The icon for 'Toggle Dark Mode' is missing; when the phone is rotated, the icon appears.


screenshot_14
screenshot_16


Best,
mrbpp

**Comment by prusnak** [comment]
> I haven't optimized the two .webp screenshots yet. ...  I'm not entirely certain about the best approach.

Do the screenshots have to be in WebP format. I did try to convert them into PNG and optimized them with the optipng script and the resulting files are much smaller:

```
 99049 screenshot_desktop.png
219531 screenshot_desktop.webp
 35446 screenshot_phone.png
 70709 screenshot_phone.webp
```

My suggestion is to use optimized PNGs if PNG format can be used.

**Comment by mrbpp** [comment]
TL;DR: Updated screenshots for desktop and phone with a classic theme, optimized sizes, reduced file size, and cleaned metadata.

I followed the instructions from the `Mozilla Developers` page. They use `.webp`, but I think `.png` will be okay. So, here is my report with `new` and improved `screenshots` from both desktop and phone:

Optimized **desktop screenshot**: 
Reduced from 220,193 to `98,187` bytes (`55.41`% decrease).
Optimized **phone screenshot**: 
Reduced from 147,160 to `120,332` bytes (`18.23`% decrease).

Additionally, I have removed the `metadata` from the **screenshots** using `exiftool`.

MAC/Homebrew:
```
exiftool -n /path/file # check metadata
exiftool -all= -overwrite_original /path/file # clean metadata
```

As you mentioned, it also worked with .png; I tested it.

scrennshot_17
scrennshot_18
scrennshot_19

I did not find any errors.

**Comment by prusnak** [comment]
Looks great.

However, the WEBP files are still being added (each file twice even).

Please remove them with:

```
git rm lnbits/static/*.webp lnbits/static/images/*.webp
git commit -a
```

Once this is addressed, we can merge IMO.

**Review by prusnak — APPROVED** [review]

**Comment by arbadacarbaYK** [comment]
Bit weird to have ANOTHER version of a logo, wonder from where that grey came. They should all be the same and not make up additional colours..

**Comment by mrbpp** [comment]
It is't another logo; it's a maskable icon. The color comes from the classic Lnbits theme.

**Comment by arbadacarbaYK** [comment]
Grey is not part of the logo. The background should not be an additional colour.

**Comment by mrbpp** [comment]
I tried with black, but it didn't look so good.

**Comment by arbadacarbaYK** [comment]
It looks exactly the way it should - like the logo. A PR about maskable icons should not mess with CI in that way.

**Comment by mrbpp** [comment]
You need to understand that the lnbits logo is a circle and some devices don't support maskable icons in the shape of a circle. I didn't want to add this additional color but the rounded square shape of the maskable icon needed a background color. I couldn't have used a black color because it would have blended in with the original color of the logo. That's why I chose the best possible option, which was the dark gray that's already been used in the lnbits theme. In my opinion, that's not a completely new logo, it was just a necessary thing to do. Thank you for the feedback and I apologize for the inconvenience.

**Comment by bitkarrot** [comment]
I'm putting my designer hat on. Design is subjective, but these icons are small and visibility at a distance is critical. The grey doesn't look great from a branding perspective. For a small icon the black should be the entire background, not circle within the bounds of the rounded icon. Perhaps this is what @arbadacarbaYK  is referring to. Example here: 

blackbkg

**Comment by mrbpp** [comment]
I like it, but I thought you wanted it to be round.

**Comment by arbadacarbaYK** [comment]
We already got 4 versions of a logo, pls just dont add another one, its all too wild. 
The background needs to be the one of the logo itself as bitkarrot has shown..

**Comment by mrbpp** [comment]
Isn't the idea to use the original LNbits logo, that's round?

**Comment by mrbpp** [comment]
Ok.

**Comment by arbadacarbaYK** [comment]
Maskable icons are all fun and games until you add a colour to it :) Doesnt matter if they are round or square

**Comment by mrbpp** [comment]
What is the color code for the dark shade in the logo (not the purple one, which represents Bolt)?

**Comment by mrbpp** [comment]
Also, @bitkarrot , what padding are you using? I'd like to replicate the same look, perhaps around 20%, but I'm not entirely sure.

**Comment by mrbpp** [comment]
**Color**: `#1e1e1e` ;
**Padding**: 40% ;

**Image Optimization Report:**

1. lnbits/static/images/maskable_icon.png
   - Original: 42098 bytes, Reduced to: 15727 bytes (62.64% decrease)

2. lnbits/static/images/maskable_icon_x192.png
   - Original: 4257 bytes, Reduced to: 2383 bytes (44.02% decrease)

3. lnbits/static/images/maskable_icon_x512.png
   - Original: 14654 bytes, Reduced to: 6144 bytes (58.07% decrease)

4. lnbits/static/images/maskable_icon_x96.png
   - Original: 2029 bytes, Reduced to: 1231 bytes (39.33% decrease)

**Comment by mrbpp** [comment]
I have made the modifications you requested. 

@arbadacarbaYK and @bitkarrot , please take a moment to review it when you have the time. 

**Have a great weekend!**

**Review by prusnak — APPROVED** [review]

**Review by dni — CHANGES_REQUESTED** [review]
Hey! Thank you for this awesome PR!

Only one issue it needs to be based on `dev` branch!
Also formatting is not write and there is a empty file in the tools directory.

I can do that for you, but i would need open a new PR and pick your commits from here.

Is that OK? or can you do that?

**Comment by mrbpp** [comment]
I will handle it. 

I am new to Git and GitHub, and that's my first commit. 

I am so happy that I can contribute to the awesome LNbits. 

By the way, I am inspired by the great work you do, @dni. Also, I want to thank @prusnak for helping me with this PR, it was a pleasure.

**Comment by mrbpp** [comment]
I'm not sure if everything is fine now, so I would appreciate it if you @dni could double-check.
